### ISSUE TITLE
Since release 0.5.0 Debugger seems to be broken in the menu

### ISSUE THREAD

**Issue by sexybiggetje** [daveleroy/sublime_debugger#119]
Today I've upgraded two machines on ST4 build 4107 on Windows to the release of Debugger 0.5.0.
After upgrading (and even restarting Sublime) the Debugger menu has all items grayed out, the console gives the following traceback;

`
reloading plugin Debugger.start
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text\Lib\python38\sublime_plugin.py", line 312, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.8/importlib/__init__.py", line 127, in import_mod
![sublimedebugger_menu](https://user-images.githubusercontent.com/60222/121689161-965cd880-cac4-11eb-9242-020843622df4.png)
ule
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 808, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\start.py", line 15, in <module>
    from .modules.commands import DebuggerCommand, DebuggerExecCommand
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\commands.py", line 6, in <module>
    from .debugger import Debugger
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\debugger.py", line 22, in <module>
    from .terminal_process import TerminalProcess
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\terminal_process.py", line 12, in <module>
    from .libs.pywinpty.st3_windows_x32.winpty import PtyProcess
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\libs\pywinpty\st3_windows_x32\winpty\__init__.py", line 11, in <module>
    from .ptyprocess import PtyProcess
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\libs\pywinpty\st3_windows_x32\winpty\ptyprocess.py", line 21, in <module>
    from .winpty_wrapper import PTY, PY2
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\libs\pywinpty\st3_windows_x32\winpty\winpty_wrapper.py", line 12, in <module>
    from .cywinpty import Agent
ImportError: DLL load failed while importing cywinpty: %1 is not a valid Win32 application.
`

Is there something I can do about this?

**Comment by daveleroy** [comment]
This seems to be caused by importing the 32 bit version of winpty when the platform is 64 bit windows. https://github.com/daveleroy/sublime_debugger/commit/04c5ddcf5489824b74f51348d928437a987ec725

I don't have a Windows machine handy to test this though.

**Comment by daveleroy** [comment]
I released Version 0.5.1 with what I believe to be the fix.

**Comment by sexybiggetje** [comment]
Thanks for the quick support, however I now get version conflict errors with the following traceback;
Which at least looks like it's referencing the x64 version now :)

`
reloading plugin Debugger.start
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text\Lib\python38\sublime_plugin.py", line 312, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.8/importlib/__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 808, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\start.py", line 15, in <module>
    from .modules.commands import DebuggerCommand, DebuggerExecCommand
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\commands.py", line 6, in <module>
    from .debugger import Debugger
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\debugger.py", line 22, in <module>
    from .terminal_process import TerminalProcess
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\terminal_process.py", line 12, in <module>
    from .libs.pywinpty.st3_windows_x64.winpty import PtyProcess
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\libs\pywinpty\st3_windows_x64\winpty\__init__.py", line 11, in <module>
    from .ptyprocess import PtyProcess
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\libs\pywinpty\st3_windows_x64\winpty\ptyprocess.py", line 21, in <module>
    from .winpty_wrapper import PTY, PY2
  File "C:\Users\<snip>\AppData\Roaming\Sublime Text\Packages\Debugger\modules\libs\pywinpty\st3_windows_x64\winpty\winpty_wrapper.py", line 12, in <module>
    from .cywinpty import Agent
ImportError: Module use of python33.dll conflicts with this version of Python.
`

**Comment by daveleroy** [comment]
Thanks for testing this.

I ended up disabling it for now. winpty is going to need to be rebuild for python 3.8 it seems.

**Comment by sexybiggetje** [comment]
The debugger now works with the 0.5.2 release, but it said that the PHP debugger needed to be installed and then gave me the following;

from: https://github.com/xdebug/vscode-php-debug/releases/latest/download/php-debug.vsix
Failed to install adapter: HTTP Error 404: Not Found

**Comment by daveleroy** [comment]
Looks like they no longer have a stable url to download the latest release.

ce44c963ac746e4ee1fc3b4fae66a7ba060a9a65 should fix this.

**Comment by sexybiggetje** [comment]
This seems to work; I do have some errors in the debug extension now. But that's unrelated to SublimeDebugger.
Thanks for your time and work, I will try and see if I can get things working further on.

Do you have ko-fi or related service?

**Comment by theLine** [comment]
I'm unable to install the PHP adapter:
```
Installing adapter: php
from: https://github.com/xdebug/vscode-php-debug/releases/latest/download/php-debug.vsix
downloading...
Failed to install adapter: HTTP Error 404: Not Found
```

it seems that with v1.15.0 the release name have changed from `php-debug.vsix` to `php-debug-{version}.vsix`.
https://github.com/xdebug/vscode-php-debug/releases/tag/v1.14.12
https://github.com/xdebug/vscode-php-debug/releases/tag/v1.15.0


is there a way to specify the path to the vsix manually?

**Comment by daveleroy** [comment]
> I'm unable to install the PHP adapter:
> 
> ```
> Installing adapter: php
> from: https://github.com/xdebug/vscode-php-debug/releases/latest/download/php-debug.vsix
> downloading...
> Failed to install adapter: HTTP Error 404: Not Found
> ```
> 
> it seems that with v1.15.0 the release name have changed from `php-debug.vsix` to `php-debug-{version}.vsix`.
> https://github.com/xdebug/vscode-php-debug/releases/tag/v1.14.12
> https://github.com/xdebug/vscode-php-debug/releases/tag/v1.15.0
> 
> is there a way to specify the path to the vsix manually?

No there is not currently a way besides adding the package manually. A fix for the path is in 0.5.3 when it hits package control.
